pylops/utils/backend.py

```diff
@@ -486,32 +486,32 @@ def get_real_dtype(dtype: DTypeLike) -> DTypeLike:
     return np.real(np.ones(1, dtype)).dtype
 
 
-def to_cupy(x: NDArray) -> NDArray:
-    """Convert x to cupy array
+def to_cupy(x: npt.ArrayLike) -> npt.ArrayLike:
+    """Convert x to cupy array if cupy is available
 
     Parameters
     ----------
-    x : :obj:`numpy.ndarray` or :obj:`cupy.ndarray`
+    x : :obj:`numpy.ndarray`, :obj:`cupy.ndarray` or :obj:`jax.Array`
         Array to evaluate
 
     Returns
     -------
-    x : :obj:`numpy.ndarray`
+    x : :obj:`numpy.ndarray`, :obj:`cupy.ndarray` or :obj:`jax.Array`
         Converted array
 
     """
     if deps.cupy_enabled:
-        if cp.get_array_module(x) == np:
+        if cp.get_array_module(x) != cp:
             x = cp.asarray(x)
     return x
 
 
-def to_numpy(x: NDArray) -> NDArray:
+def to_numpy(x: npt.ArrayLike) -> NDArray:
```

Comment:
same

Reply:
well technically we should create CuPy/JAX types but I haven't got around that yet.. for another PR 😉